src/Domain/LeanCode.DomainModels/Ulids/Ulid.cs

```diff
@@ -0,0 +1,884 @@
+
+        return true;
+    }
+
+    public bool TryWriteStringify(Span<char> span)
+    {
+        if (span.Length < 26)
+        {
+            return false;
+        }
+
+        span[25] = Base32Text[randomness9 & 31]; // eliminate bounds-check of span
+
+        // timestamp
+        span[0] = Base32Text[(timestamp0 & 224) >> 5];
+        span[1] = Base32Text[timestamp0 & 31];
+        span[2] = Base32Text[(timestamp1 & 248) >> 3];
+        span[3] = Base32Text[((timestamp1 & 7) << 2) | ((timestamp2 & 192) >> 6)];
+        span[4] = Base32Text[(timestamp2 & 62) >> 1];
+        span[5] = Base32Text[((timestamp2 & 1) << 4) | ((timestamp3 & 240) >> 4)];
+        span[6] = Base32Text[((timestamp3 & 15) << 1) | ((timestamp4 & 128) >> 7)];
+        span[7] = Base32Text[(timestamp4 & 124) >> 2];
+        span[8] = Base32Text[((timestamp4 & 3) << 3) | ((timestamp5 & 224) >> 5)];
+        span[9] = Base32Text[timestamp5 & 31];
+
+        // randomness
+        span[10] = Base32Text[(randomness0 & 248) >> 3];
+        span[11] = Base32Text[((randomness0 & 7) << 2) | ((randomness1 & 192) >> 6)];
+        span[12] = Base32Text[(randomness1 & 62) >> 1];
+        span[13] = Base32Text[((randomness1 & 1) << 4) | ((randomness2 & 240) >> 4)];
+        span[14] = Base32Text[((randomness2 & 15) << 1) | ((randomness3 & 128) >> 7)];
+        span[15] = Base32Text[(randomness3 & 124) >> 2];
+        span[16] = Base32Text[((randomness3 & 3) << 3) | ((randomness4 & 224) >> 5)];
+        span[17] = Base32Text[randomness4 & 31];
+        span[18] = Base32Text[(randomness5 & 248) >> 3];
+        span[19] = Base32Text[((randomness5 & 7) << 2) | ((randomness6 & 192) >> 6)];
+        span[20] = Base32Text[(randomness6 & 62) >> 1];
+        span[21] = Base32Text[((randomness6 & 1) << 4) | ((randomness7 & 240) >> 4)];
+        span[22] = Base32Text[((randomness7 & 15) << 1) | ((randomness8 & 128) >> 7)];
+        span[23] = Base32Text[(randomness8 & 124) >> 2];
+        span[24] = Base32Text[((randomness8 & 3) << 3) | ((randomness9 & 224) >> 5)];
+
+        return true;
+    }
+
+    public override string ToString()
+    {
+        return string.Create(26, this, (span, ulid) => ulid.TryWriteStringify(span));
+    }
+
+    public string ToString(string? format, IFormatProvider? formatProvider) => ToString();
+
+    public bool TryFormat(
+        Span<char> destination,
+        out int charsWritten,
+        ReadOnlySpan<char> format,
+        IFormatProvider? provider
+    )
+    {
+        if (TryWriteStringify(destination))
+        {
+            charsWritten = 26;
+            return true;
+        }
+        else
+        {
+            charsWritten = 0;
+            return false;
+        }
+    }
+
+    // Comparable/Equatable
+
+    public override unsafe int GetHashCode()
+    {
+        // Simply XOR, same algorithm of Guid.GetHashCode
+        fixed (void* p = &this.timestamp0)
```

Comment:
Uhh okay, this is a bit mindbending. I think we could get away with `Unsafe.ReadUnaligned(Unsafe.Add(…))` but I'll have to verify that.